x-pack/legacy/plugins/spaces/server/routes/api/v1/spaces.ts

```diff
@@ -5,16 +5,49 @@
  */
 
 import Boom from 'boom';
+import { Legacy } from 'kibana';
 import { Space } from '../../../../common/model/space';
 import { wrapError } from '../../../lib/errors';
 import { SpacesClient } from '../../../lib/spaces_client';
 import { addSpaceIdToPath } from '../../../lib/spaces_url_parser';
 import { getSpaceById } from '../../lib';
 import { InternalRouteDeps } from '.';
+import { getActiveSpace } from '../../../lib/get_active_space';
+import { getSpaceSelectorUrl } from '../../../lib/get_space_selector_url';
 
 export function initInternalSpacesApi(deps: InternalRouteDeps) {
   const { http, config, spacesService, savedObjects, routePreCheckLicenseFn } = deps;
 
+  http.route({
+    method: 'GET',
+    path: '/api/spaces/v1/activeSpace',
+    async handler(request: Legacy.Request) {
+      const spacesClient: SpacesClient = await spacesService.scopedClient(request);
+      return await getActiveSpace(
+        spacesClient,
+        http.basePath.get(request),
+        config.get('server.basePath')
+      );
+    },
+    options: {
+      pre: [routePreCheckLicenseFn],
+    },
+  });
+
+  http.route({
+    method: 'GET',
+    path: '/api/spaces/v1/npStart',
+    async handler() {
+      return {
+        spaceSelectorUrl: getSpaceSelectorUrl(config),
+      };
+    },
+    options: {
+      auth: false,
```

Comment:
This is called as part of plugin startup, which can happen on unauthenticated routes.
It is effectively returning `server.basePath`, which is already immediately obvious to anyone navigating to Kibana when unauthenticated, so there is no danger of having this run without authentication.